dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java

```diff
@@ -122,17 +127,128 @@ public void cleanAfterMethodInvocation() {
+     */
     @Override
     public Object getInjectionObject(Object targetObject,
-                                     InjectionTargetContext targetContext) throws InjectionException {
-        try {
-            return super.getInjectionObject(targetObject, targetContext);
-        } catch (InjectionException e) {
-            if (!CDIUtils.isInjectionFailureIgnored()) {
-                throw e;
-            } else {
+            InjectionTargetContext targetContext)
+                    throws InjectionException {
+
+        Object retObj = null;
+
+        try
+        {
+            retObj = getInjectionObjectInstance(targetObject, targetContext);
+        } catch (RecursiveInjectionException ex)
+        {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled())
+                Tr.exit(tc, "getInjectionObject: " + ex);
+            throw ex;
+        } catch (Throwable ex)
+        {
+
+            if (CDIUtils.isInjectionFailureIgnored()){
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "com.ibm.ws.cdi.ignoreInjectionFailure = true, ignoring InjectionException", ex);
+                }
+                return null;
+            }
+
+            // Only log FFDC if this is a failed injection; failed naming lookups may
+            // be normal and the caller should decide if FFDC is needed.
+            if (targetObject != null) {
+                FFDCFilter.processException(ex, CLASS_NAME + ".getInjectionObject",
+                        "408", this, (Object[]) null);
+            }
+
+            String displayName = getDisplayName();
+            Object exMessage = ex.getLocalizedMessage();
+            if (exMessage == null)
+            {
+                exMessage = ex.toString();
+            }
+
+            String message = Tr.formatMessage(tc,
+                                                  "FAILED_TO_CREATE_OBJECT_INSTANCE_CWNEN0030E",
+                                                  displayName,
+                                                  exMessage);
+            InjectionException ex2 = new InjectionException(message, ex);
+
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled())
+                Tr.exit(tc, "getInjectionObject", ex2);
+            throw ex2;
+        }
+        
+        if (retObj == null) 
+        {
+            String classTypeName, component, module, app;
+            if (ivJ2eeName != null)
+            {
+                component = ivJ2eeName.getComponent();
+                module = ivJ2eeName.getModule();
+                app = ivJ2eeName.getApplication();
+            }
+            else
+            {
+                component = module = app = "UNKNOWN";
+            }
+
+            classTypeName = getInjectionClassType() == null ? "UNKNOWN" : getInjectionClassType().getName();
+
+            Tr.error(tc, "UNABLE_TO_RESOLVE_INJECTION_OBJECT_CWNEN0035E",
```

Comment:
This error message is NLS'd but the exception message below is not. The exception message should be NLS'd

Reply:
This one pre-dates me. But I can add a comment noting this.